src/main/java/com/ldtteam/structurize/management/StructureName.java

```diff
@@ -106,13 +109,18 @@ else if (name.startsWith(Structures.SCHEMATICS_CACHE + Structures.SCHEMATICS_SEP
         {
             hut = schematic.split("\\d+")[0];
             section = Structures.SCHEMATICS_PREFIX;
-            if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(Constants.MINECOLONIES_MOD_ID, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
+            //This allows mods other than minecolonies to create hut buildings and use the buildtool with them.
+            for (final Map.Entry<String, ModFileInfo> origin : StructureLoadingUtils.getOriginMods().entrySet())
```

Comment:
You should be fine with key view (you don't seem to use value from entry set)

Reply:
You are absolutely right.